src/catchup/simulation/GenerateBucketsWork.h

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#pragma once
+
+#include "history/HistoryArchive.h"
+#include "work/BasicWork.h"
+
+namespace stellar
+{
+
+class Bucket;
+class TmpDir;
+
+/*
+ * GenerateBucketsWork transforms the existing bucketlist by creating new ledger
+ * entries that mimic the existing ones. This is useful for simulation and test
+ * purposes, as production buckets can be used to create a more or less
+ * plausible set of LedgerEntries and simulate a much bigger ledger. This work
+ * assumes presence of the bucket files in BUCKET_DIR_PATH. Each bucket is
+ * transformed into a new, bigger bucket that mimics lifecycles of given ledger
+ * entries, but with artificially generated account keys.
+ * */
+
+class GenerateBucketsWork : public BasicWork
+{
+    Application& mApp;
+    HistoryArchiveState const mApplyState;
+
+    // New HAS is populated incrementally
+    HistoryArchiveState mGeneratedApplyState;
+    uint32_t const mMultiplier;
+    uint32_t mLevel;
+
+    std::shared_ptr<Bucket> mPrevSnap;
+
+    // Keep references to created buckets for cleanup
+    std::vector<std::shared_ptr<Bucket>> mBuckets;
```

Comment:
Agreed that the comment isn't super clear. The reason I added this is to ensure downstream works can't forget newly created buckets (actually ran into this issue, since `GenerateBucketsWork` populates a HAS, which only keeps bucket hash strings). Re-visiting this, I've made two changes: 1. update the comment to make it clear why these buckets need to remembered, and 2. pass a reference to the generated buckets map to `GenerateBucketsWork` to make the ownership a bit more explicit (i.e. let `SimulateApplyBucketsWork` track generated buckets)